ocaml/xapi/extauth_ad.ml

```diff
@@ -0,0 +1,74 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+(**
+ * @AD group Access Control
+*)
+
+open Db_actions
+
+module D = Debug.Make (struct let name = "extauth_ad" end)
+
+open D
+
+exception Unknown_Ad_type of string
```

Comment:
I would suggest to spell this consistently as `AD` where possible; maybe `Unknown_AD_type`. I believe this should be declared in the module below.